cpp/src/arrow/filesystem/hdfs_io.h

```diff
@@ -38,18 +40,6 @@ namespace io {
 class HdfsReadableFile;
```

Comment:
Most of these declarations should IMHO go into the `arrow::filesystem::internal` namespace, except for `HdfsConnectionConfig` which can go into `arrow::filesystem`.